catalog/security/banner-marking/src/test/groovy/org/codice/alliance/security/banner/marking/PortionMarkingsSpec.groovy

```diff
@@ -0,0 +1,632 @@
+
+class PortionMarkingsSpec extends Specification {
+    def 'test type and classification'() {
+        when:
+        def portionMarkings = PortionMarkings.parseMarkings(markings)
+
+        then:
+        portionMarkings.type == type
+        portionMarkings.classification == classification
+
+        where:
+        markings                         | type  | classification
+        'TS'                             | US    | TOP_SECRET
+        'S//FGI FRA'                     | US    | SECRET
+        'C'                              | US    | CONFIDENTIAL
+        '//CAN R'                        | FGI   | RESTRICTED
+        '//CTS//BOHEMIA'                 | FGI   | TOP_SECRET
+        '//JOINT S CAN USA'              | JOINT | SECRET
+        '//JOINT R CAN GBR'              | JOINT | RESTRICTED
+        '//JOINT TS CAN DEU USA'         | JOINT | TOP_SECRET
+    }
+
+    def 'test fgi markings'() {
+        when:
+        def portionMarkings = PortionMarkings.parseMarkings(markings)
+
+        then:
+        portionMarkings.type == FGI
+        portionMarkings.classification == classification
+        portionMarkings.fgiAuthority == fgiAuthority
+
+        where:
+        markings                | classification | fgiAuthority
+        '//NS//ATOMAL'          | SECRET         | 'NATO'
+        '//DEU R'               | RESTRICTED     | 'DEU'
+    }
+
+    def 'test joint markings'() {
+        when:
+        def portionMarkings = PortionMarkings.parseMarkings(markings)
+
+        then:
+        portionMarkings.type == JOINT
+        portionMarkings.classification == classification
+        portionMarkings.jointAuthorities.containsAll(jointAuthorities)
+
+        where:
+        markings                                      | classification | jointAuthorities
+        '//JOINT S CAN USA'                           | SECRET         | ['CAN', 'USA']
+        '//JOINT TS CAN DEU USA'                      | TOP_SECRET     | ['CAN', 'DEU', 'USA']
+        '//JOINT S CAN DEU USA NATO//TK//RELIDO'      | SECRET         | ['CAN', 'DEU', 'USA', 'NATO']
+    }
+
+    def 'U//FOUO'() {
+        when:
+        def portionMarkings = PortionMarkings.parseMarkings("U//FOUO")
+
+        then:
+        portionMarkings.classification == UNCLASSIFIED
+        portionMarkings.disseminationControls.contains(FOUO)
+    }
+
+    def 'test sci controls only'() {
+        when:
+        def portionMarkings = PortionMarkings.parseMarkings(markings)
+
+        then:
+        portionMarkings.classification == classification
+        portionMarkings.sciControls.size() == sciControl.size()
+
+        for (SciControl control : portionMarkings.sciControls) {
+            def expectedControl = sciControl.stream()
+                    .filter({ c -> c.control == control.control })
+                    .findFirst().orElse(null)
+
+            // When looping, Spock gets confused without using an explicit 'assert' statement
```

Comment:
Maybe a little pedantic, but spock is not getting confused within here - it is working as designed.

http://spockframework.org/spock/docs/1.0/spock_primer.html?when_and_then_blocks#_implicit_and_explicit_conditions

https://github.com/spockframework/spock/issues/410#issuecomment-136152439

Reply:
Removed the comment.